integration-client/src/main/java/com/lightbend/lagom/javadsl/client/integration/LagomClientFactory.java

```diff
@@ -198,21 +203,28 @@ public void addStopHook(Callable<? extends CompletionStage<?>> hook) {
             }
         }, actorSystem.dispatcher());
 
+        ServiceInfo serviceInfo = new ServiceInfo(serviceName);
+
+        // Kafka client
+        TopicFactory kafkaTopicFactory = new KafkaTopicFactory(serviceInfo, actorSystem, materializer,
+                actorSystem.dispatcher());
+        TopicFactoryProvider topicFactoryProvider = () -> Some.apply(kafkaTopicFactory);
```

Comment:
Yes it does. I don't think making it configurable should be in scope right now, the main problem is that this client is designed to be lightweight - there's no injector created for example. The big question is how would you get shared dependencies (such as the actor system) into the client? If we had an injector, that'd be simple, get the class by the injector and let the injector work out how to create it. But since we don't have that, it's not straight forward. I think in this case it would be better to wait for the use cases to come up to work out what needs to be done, then to try and design an API now.